### PR TITLE
Set reasonable defaults for pandas dataframe default renderer

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
pandas dataframe default renderer should be cheap to render

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
The [default deck renderer for pandas dataframes](https://github.com/flyteorg/flytekit/blob/master/flytekit/deck/renderer.py#L27) relies on pandas dataframe [to_html](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_html.html), but big dataframes take a long time to return, so in this PR we set reasonable defaults for both the max number of rows and columns.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

**Review by pingsutw — DISMISSED** [review]

File: flytekit/deck/renderer.py [hunk 1]
```diff
@@ -14,17 +14,22 @@ def to_html(self, python_value: Any) -> str:
...
+DEFAULT_MAX_COLS = 100
+
+
 class TopFrameRenderer:
     """
     Render a DataFrame as an HTML table.
     """
 
-    def __init__(self, max_rows: Optional[int] = None):
+    def __init__(self, max_rows: Optional[int] = DEFAULT_MAX_ROWS, max_cols: Optional[int] = DEFAULT_MAX_COLS):
```
Comment (wild-endeavor) [inline flytekit/deck/renderer.py:26]: not optional if you give a default.
```suggestion
    def __init__(self, max_rows: int = DEFAULT_MAX_ROWS, max_cols: int = DEFAULT_MAX_COLS):
```

Comment (wild-endeavor) [reply flytekit/deck/renderer.py:26]: right?  or not?  i never figured out how this works?  i think it's from the perspective of the body of the function right?  not the caller

**Review by wild-endeavor — DISMISSED** [review]

File: flytekit/deck/renderer.py (hunk 1 above)
Comment (eapolinario) [reply flytekit/deck/renderer.py:26]: that's true, we only need `Optional` if we want to explicitly support passing `None`, which is not the case here.

**Review by wild-endeavor — APPROVED** [review]
